draft-ietf-cdni-ci-triggers-rfc8007bis.xml

```diff
@@ -2883,7 +2889,125 @@ https://video.example.com/k/movie1/4/013.ts
          ]]></artwork>
           </figure>
         </section>
-      </section>
+	</section>
+	<section anchor="execution-policy" title="Execution Extension">
+	<t>
+		Unless specified otherwise, dCDN is at liberty to decide how to choose trigger commands for execution from all pending commands, whether to process trigger commands sequentially or in parallel, immediately upon acceptance, or with a delay in batches. A uCDN may wish to control trigger processing in more detail, including order of execution, dependencies, and concurrency.
+	</t>
+	<t>
+	Example use cases:
+		<list style="symbols">
+		<t>
+		Priority:  uCDN may have multiple trigger commands in “pending” and/or “active” mode. The trigger commands may potentially take a long time to execute, for example due to policy constraints, large number of content objects affected by it or other dCDN business logic.
+		uCDN may wish to prescribe the order in which dCDN picks up its trigger commands for execution from the “pending” queue, by indicating a relative priority of each trigger. The priority would affect selection of trigger commands specific to the requesting uCDN. dCDN may separately prioritize triggers from multiple uCDNs subject to its business logic. 
+
+		Multiple priority-related sub use cases exist: 
+		<list style="symbols">
+		<t> (i) uCDN needs to introduce urgent “purge” or “invalidate” trigger into an existing queue of trigger commands to correct wrong versions of content objects published by it</t>
+		<t> (ii) uCDN needs to indicate which content objects should be prepositioned, purged or invalidated first, for example prepositioning newer released content before prepositioning updates to existing catalog</t>
+		</list>
+		</t>
+
+		<t>
+		Pre-requisite:  In some cases uCDN may wish to indicate what trigger commands should be processed and completed before another trigger command is processed. 
+			For example, uCDN may want to rectify incorrectly published content by purging content objects and then prepositioning them again. In this case, uCDN may want the preposition trigger command to be processed only after the purge trigger command has been processed because concurrent processing of these triggers may cause the new version of these content objects to be purged. Alternatively, uCDN may wish to avoid processing of purge and invalidation triggers in parallel to long-running preposition triggers that could lead to race conditions, and condition execution of such purge or revalidation triggers upon completion or cancellation of the pre-positioning triggers. 
```

Comment:
The last sentence is a little wordy and difficult to read.  Maybe:  "Alternatively, uCDN may wish to condition the execution of purge or invalidation triggers upon the completion or cancellation of long-running preposition triggers to avoid race conditions that would result from processing these in parallel."

Reply:
Accepted